modules/apps/frontend-data-set/frontend-data-set-taglib/src/main/java/com/liferay/frontend/data/set/taglib/internal/jaxrs/application/FDSApplication.java

```diff
@@ -179,6 +183,39 @@ public Response saveActiveFDSViewSettings(
 		).build();
 	}
 
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Path("/fds/{fdsName}/custom-views")
+	@POST
+	public Response saveFDSCustomView(
+		@PathParam("fdsName") String fdsName,
+		@Context HttpServletRequest httpServletRequest, String bodyJSON) {
```

Comment:
I think here you can define a class for the bodyJSON and JAX-RS would handle the deserialization and serialization making it easier

Reply:
Just to give you a little more context. If you create a class FDSCustomView with the necessary fields and you use it in the method signature like
```
public FDSCustomView saveFDSCustomView(
		@PathParam("fdsName") String fdsName,
		@Context HttpServletRequest httpServletRequest, FDSCustomView) 
```

But thinking about it. As we are using a JAX-RS application from the -taglib module and we don't use the Liferay REST infrastructure we might have some problems...

I don't know if it worth give it a try, or we can leave this refactor pending until we extract the code to a REST API module

Reply:
OK, let's leave it for a separate task. I added this as well to the improvements list. 

Sidenote - there is also[ `activeViewSettingsJSON`](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-taglib/src/main/java/com/liferay/frontend/data/set/taglib/internal/jaxrs/application/FDSApplication.java#L141) where we could apply the same change, but we will likely remove that endpoint soon.

Reply:
we can change the endpoint implementation later on (i.e. in a separate task or even epic)